File: Offer/2277/main.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i, j = 0, len(nums) - 1
        temp: int = 0
        while i < j:
            temp = nums[i] + nums[j]
            if temp < target:
                i += 1
                continue
            if temp > target:
                j -= 1
                continue
            return [nums[i], nums[j]]
        return []
                

File: Offer/2277/main.go
package main

func twoSum(nums []int, target int) []int {
	i, j := 0, len(nums)-1
	var temp int
	for i < j {
		temp = nums[i] + nums[j]
		if temp < target {
			i++
			continue
		}
		if temp > target {
			j--
			continue
		}
		return []int{nums[i], nums[j]}
	}
	return []int{}
}
